lib/context.js

```diff
@@ -84,3 +84,24 @@ Context.prototype.retries = function (n) {
   this.runnable().retries(n);
   return this;
 };
+
+/**
+ * Defines a global read-only property `mochaVar` that provides access to Mocha's name and version.
+ * It is accessible globally within the Node.js environment or in the browser
+ * @example
+ * console.log(globalThis.mochaVar); // Outputs: { name: "mocha", version: "X.Y.Z" }
+ *
+ * @property {Object} mochaVar - The global property containing Mocha's name and version.
+ * @property {string} mochaVar.name - The name of the Mocha package.
+ * @property {string} mochaVar.version - The current version of the Mocha package.
+ */
+var mochaPackageJson = require('../package.json');
+var version = mochaPackageJson.version;
+var name = mochaPackageJson.name;
+
+Object.defineProperty(globalThis, 'mochaVar', {
+  get: () => ({
+    name,
+    version
+  })
+});
```

Comment:
This is fine for most JS environments, but some sandboxed ones disallow changing `globalThis` and might even throw. https://github.com/endojs/endo exists today; https://github.com/tc39/proposal-shadowrealm will hopefully exist eventually.

Let's wrap this in a try/catch the same way `Error.stackTraceLimit = Infinity;` is wrapped:

https://github.com/mochajs/mocha/blob/3f13bbcd4bb96d830f6eeb67f5b6bcceffb8f93b/lib/cli/cli.js#L43-L47